mod_reforged/hooks/skills/special/no_ammo_warning.nut

```diff
@@ -0,0 +1,17 @@
+
+::mods_hookExactClass("skills/special/no_ammo_warning", function(o) {
+    // Temporarily changes the getID function from the ammo item into getAmmoType
+    local isHidden = o.isHidden;
+    o.isHidden = function()
+    {
+        local ammo = this.getContainer().getActor().getItems().getItemAtSlot(::Const.ItemSlot.Ammo);
+        if (ammo == null) return isHidden();
+        local oldGetID = ammo.item.getID;
+        ammo.item.getID = ammo.getAmmoType;
+
+        local ret = isHidden();
+
+        ammo.item.getID = oldGetID;
```

Comment:
Not the *biggest* fan of this, I think it'd be better if we also handle the situation where someone creates a child getID function, but if you don't want to put in the work I understand since it is fairly niche.

Although reading back, you do this differently here than in actor.nut getRangedWeaponInfo, which is inconsistent.

Reply:
I haven't considered the cases with childs having that function at all. Not sure how to best deal with that.

About the second issue: I revisited those three hooks and unified them now